hawkbit-extension-artifact-repository-s3/src/main/java/org/eclipse/hawkbit/artifact/repository/S3Repository.java

```diff
@@ -179,22 +179,26 @@ public AbstractDbArtifact getArtifactBySha1(final String tenant, final String sh
         final String key = objectKey(tenant, sha1Hash);
 
         LOG.info("Retrieving S3 object from bucket {} and key {}", s3Properties.getBucketName(), key);
-        try (final S3Object s3Object = amazonS3.getObject(s3Properties.getBucketName(), key)) {
-            if (s3Object == null) {
-                return null;
-            }
+        final S3Object s3Object = amazonS3.getObject(s3Properties.getBucketName(), key);
```

Comment:
I think you missed that this is a try-resource case which automatically will close the `s3Object` - so you don't need to split the try-resource in a try-finally block, will do the same.

This has been added in the commit : https://github.com/eclipse/hawkbit/commit/c3908c9c3458387a6cb7d3e776d779bbec1a4aaa as this was definetly a memory leak problem.